cpp/oneapi/dal/algo/pca/detail/train_ops.hpp

```diff
@@ -39,28 +39,31 @@ struct train_ops {
     using result_t = train_result<task_t>;
     using descriptor_base_t = descriptor_base<task_t>;
 
-    void check_preconditions(const Descriptor& params, const input_t& input) const {
+    void check_preconditions(const Descriptor& desc, const input_t& input) const {
         using msg = dal::detail::error_messages;
 
         if (!input.get_data().has_data()) {
             throw domain_error(msg::input_data_is_empty());
         }
-        if (input.get_data().get_column_count() < params.get_component_count()) {
+        if (input.get_data().get_column_count() < desc.get_component_count()) {
```

Comment:
This condition seems to be incorrect since `dest.get_component_count()` could be `0` which means we should task all the columns of input data for principal components. I think we need to add a special test case for component count equal to `0`

Reply:
There is a test case for `component_count == 0`. 

Condition seems correct for `component_count == 0`, because `data.column_count < 0` is always `false`:
`table` invariant `data.column_count >= 0` => `0 < 0 == false`